### PR TITLE
[Impeller] Add gradient dithering for Radial/Sweep/Conical gradients

### PR DESCRIPTION
**Pull request by matanlurey**
Partial work towards https://github.com/flutter/flutter/issues/131450 similar to https://github.com/flutter/engine/pull/44181.

---

Run the Playground locally:

```bash
$ENGINE/out/host_debug_unopt/impeller_unittests \
  --enable_playground \
  --gtest_filter="*CanRender*GradientWithDithering*"
```

### Before

Screenshot 2023-08-03 at 10 08 53 AM

### After

Screenshot 2023-08-03 at 10 13 57 AM

### Before

Screenshot 2023-08-03 at 10 27 35 AM

### After

Screenshot 2023-08-03 at 10 33 11 AM

### After

Screenshot 2023-08-08 at 1 11 40 PM

### CODE REVIEW

**Comment by jonahwilliams** [comment]
I think we have a mysterious 4th gradient type, but maybe that is combined into the radial gradient? The two point conical gradient.

**Comment by jonahwilliams** [comment]
This one: https://github.com/flutter/engine/blob/main/impeller/entity/shaders/conical_gradient_fill.frag

File: impeller/aiks/aiks_unittests.cc [hunk 1]
```diff
@@ -449,6 +449,66 @@ TEST_P(AiksTest, CanRenderLinearGradientWithDitheringEnabled) {
   CanRenderLinearGradientWithDithering(this, true);
 }  // namespace
 
+namespace {
```
Comment (chinmaygarde) [inline impeller/aiks/aiks_unittests.cc:452]: Just `static` instead of the anonymous namespace?

**Review by chinmaygarde — APPROVED** [review]

File: impeller/aiks/aiks_unittests.cc (hunk 1 above)
Comment (matanlurey) [reply impeller/aiks/aiks_unittests.cc:452]: Done. Also applied to my other gradient test cases since they are quite similar.

**Comment by matanlurey** [comment]
> I think we have a mysterious 4th gradient type, but maybe that is combined into the radial gradient? The two point conical gradient.

Thanks! Added.

**Review by jonahwilliams — APPROVED** [review]
LGTM

**Review by jonahwilliams — COMMENTED** [review]
> There was a compiler error: Unsigned integers are not supported on legacy ESSL.

Huh

File: impeller/entity/shaders/conical_gradient_fill.frag [hunk 2]
```diff
@@ -45,4 +47,8 @@ void main() {
                                  frag_info.tile_mode,                      //
                                  frag_info.decal_border_color);
   frag_color = IPPremultiply(frag_color) * frag_info.alpha;
+
+  if (frag_info.dither) {
```
Comment (jonahwilliams) [inline impeller/entity/shaders/conical_gradient_fill.frag:51]: This should be in the _ssbo varaint

**Pull request merged**